cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -228,6 +231,31 @@ public BigDecimal getDecimal(String fieldName) {
     }
   }
 
+  /**
+   * Returns value of a datetime field as java.time.LocalDateTime formatted using
+   * {@link java.time.format.DateTimeFormatter#ISO_LOCAL_DATE_TIME}. Use get() to retrieve value as string.
+   *
+   * @param fieldName Name of the field
+   * @return {@link LocalDateTime}
+   *
+   */
+  @Nullable
+  public LocalDateTime getDateTime(String fieldName) {
```

Comment:
Does the datetime type we are adding always not supporting Zone? Otherwise, why use LocalDateTime instead of ZonedDateTime?

Reply:
BQ and a few databases that I checked uses DateTime for storing DateTime without the offset/zone. Following the same here. https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#datetime_type

Reply:
The question is, if the incoming datatime string has timezone in it (e.g. `2021-01-27T12:24:36+08:00`), what are we going to do with it? Is it getting thrown away?

Reply:
Yes, we are throwing an UnexpectedFormatException for that. There are a few Wrangler directives planned for converting different format strings and also to convert from timestamp using a timezone.

Reply:
But why we don't want to support it in the core Schema class?

Reply:
Zone information will not be available for some sources like BQ DateTime. In that case , do we assume some default zone? Also while writing to BQ sink, we will have to remove the zone information.

Reply:
In ISO-8861, if no timezone is provided, the zone is local time zone (which is just like what you have in this PR). If zone information is provided, then it is as defined by the string.

Reply:
Discussed offline and decided to support Datetime without offset in the first version. If required this can be extended to support zone information or add a new datatype that supports datetime with zone.